### DISCUSSION TITLE
B站直播间换真·原画源

### DISCUSSION THREAD

**Discussion by caleyuexing** [the1812/Bilibili-Evolved#4907] [功能建议 / Ideas]
**B站在当主播刚开播的时候给观众分配一个live_[UID]_[数字]/index.m3u8源，这个是真正的原画。但当直播间人数变多后会变成live_[UID]_[数字]_bluray/index.m3u8,这个源明显是经过二压的。**

**我尝试过当二压条件触发后在本地播放器上用真·原画源，是可用并且画质也更好的，本人学的不是很多，不知道能不能直接在B站直播页面换源，求大佬解答**

图示：左为真·原画源；右为伪·原画源

![微信图片_20240911222823](https://github.com/user-attachments/assets/d01d376c-926d-49f3-8b9d-acafaed128a8)

**Comment by DecCapri** [comment]
这么神奇的吗

**Comment by CloudyEagle** [comment]
我曾经从事过做B站直播存档的事，还是追求主播开播后最原始未压制的直播流。
自从B站直播对原画流开始二次压制，就很难观看和录制到了，观看一般是要在开播前候场能看到；
录制即使开播前开启录制软件，甚至部署3套以上的直播流录制软件才能保存到完整一整场的。
因为有些直播的cdn稳定性堪忧，就算是连到真原画源，有时会抽风断流，再就切换到伪原画源，提供观看或者录制。
据说真原画直播流在伪原画直播流出来后，在一段时间内是有效的，有人说是半小时，有人说是1小时，
可能还是要看主播的同接量和热度，链接的时效性过后就无法获取到内容了。
具体可以看下这款[录播软件](https://github.com/BililiveRecorder/BililiveRecorder)里的issue中曾经的讨论，可能已关闭。
